Add bucket options strings set

keithele opened the pull request:
This update includes the new "options" strings set returned by the B2 API in the Bucket object.

ppolewicz commented:
Options don't seem to be documented in the API docs (yet?)

keithele commented:
@ppolewicz Ah, you're right.  It's not in the B2 API documentation yet.  It is being returned by the API in production right now, so wanted to update the SDK to handle it.  I've made a note to update the docs.  Thanks!

ppolewicz commented:
My point is that without the docs an additional field cannot really be used as nobody knows how to interpret the returned data. Is the resulting option set made of a limited number of strings that have some additional meaning? Because if that is true, we should parse them into a series of classes with properties that describe the behavior of the options to the SDK user to interpret them.

Can we wait with merging this until it is documented, so that your patch can be cross-reviewed?

keithele commented:
I've updated the B2 API documentation.  The `options` parameter doesn't have any values for now.  When strings are added in the future, then the possible values can be documented and added for use in the SDK.  But for now, there aren't any values to document, it's a future-looking/flexible set of strings.  Does that help?

ppolewicz commented:
Can you give an example of how it might be used?

keithele commented:
I don't have an example because there aren't any uses of it.  Just matching the current API.

ppolewicz commented on test/v0/test_api.py:
```diff
@@ -38,6 +38,7 @@ def test_list_buckets(self):
             'bucketInfo': {},
             'corsRules': [],
             'lifecycleRules': [],
+            'options': set(),
```
This change should be done in `test/v1/test_api.py:test_list_buckets` too

keithele replied:
Unfortunately, the `delete_bucket` api being used by `test/v1/test_api.py:test_list_buckets` doesn't have a return value/it's `None`.  So the existing test case is valid according to the existing API in `/api.py`.

Would we want to change `/api.py:delete_bucket()` to return the dictionary from `/raw_api.py:delete_bucket()`?  
Was there a reason the old API returned the value from `delete_bucket` and the new one doesn't?  What do you think?

ppolewicz replied:
Oh, you are right!

I think it was listed in a comment as something that is not needed. Should the user of `delete_bucket()` expect to receive a dict/json in return to his (successful) call? Why?

keithele replied:
Looks like you updated it back in June.  I think it works fine as-is.  If you're good with it we can mark this resolved and merge.
